r2-shared/Publication/Locator.ts

```diff
@@ -0,0 +1,34 @@
+interface LocatorText {
+  after?: string;
+  before?: string;
+  highlight?: string;
+}
+
+interface StartEndInfo {
+  cssSelector: string;
+  textNodeIndex: number;
+  charOffset?: number;
+}
+
+interface DomRangeInfo {
```

Comment:
Also the name of these structures is a bit outdated I think, it should be `DOMRange` : https://github.com/readium/architecture/blob/master/models/locators/extensions/html.md#the-start-and-end-object

for `StartEndInfo` there's no clear naming, but we used `DOMRange.Point` on mobile, since it fits the description: "A serializable representation of the "start" **boundary point** of the DOM Range"